prow/gerrit/reporter/reporter_test.go

```diff
@@ -172,23 +251,99 @@ func TestReport(t *testing.T) {
 				},
 				Status: v1.ProwJobStatus{
 					State: v1.SuccessState,
-					URL:   "guber/foo/bar",
+					URL:   "guber/foo",
 				},
 				Spec: v1.ProwJobSpec{
 					Refs: &v1.Refs{
-						Repo: "foo/bar",
+						Repo: "foo",
+					},
+					Job: "ci-foo",
+				},
+			},
+			existingPJs: []*v1.ProwJob{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"gerrit-revision": "abc",
+						},
+						Annotations: map[string]string{
+							"gerrit-id":       "123-abc",
+							"gerrit-instance": "gerrit",
+						},
+					},
+					Status: v1.ProwJobStatus{
+						State: v1.PendingState,
+						URL:   "guber/bar",
+					},
+					Spec: v1.ProwJobSpec{
+						Refs: &v1.Refs{
+							Repo: "bar",
+						},
+						Job: "ci-bar",
+					},
+				},
+			},
+		},
+		{
+			name: "finished pj, with other job on same revision, finished",
```

Comment:
finished pj with other job on same revision, finished. both passing. should comment

Might want to do all of these scenarios:

* failure => failure + failure
* failure => failure + pass
* pass => pass + pass
* pass => pass + failure

Reply:
covered pass, fail, pass+fail, pass+pass now